compair/static/modules/course/course-module.js

```diff
@@ -346,14 +352,33 @@ module.controller(
                         return !assignment.available;
                     // STUDENTS: return all assignments that need to be answered or compared
                     case "My unfinished assignments":
-                        return (assignment.answer_period && !assignment.status.answers.answered) ||
-                            (assignment.compare_period && assignment.steps_left > 0);
+                        return (permissions.canAnswer && permissions.needsAnswer) ||
+                            (permissions.canCompare && permissions.needsCompareOrSelfEval);
+                    // STUDENTS: return all assignments that have a saved draft answer, comparison, or self-eval
+                    case "Drafts saved":
+                        return (permissions.canAnswer && permissions.needsAnswer && permissions.hasDraftAnswer) ||
+                            (permissions.canCompare && permissions.needsCompare && permissions.hasDraftComparison) ||
+                            (permissions.canCompare && permissions.needsSelfEval && permissions.hasDraftSelfEval);
+                    // STUDENTS: return all assignments that have received feedback
+                    case "Feedback received":
+                        return permissions.hasFeedback;
                     default:
                         return false;
                 }
             }
         }
 
+        $scope.assignmentSortOrder = function(sortOrder) {
+            switch (sortOrder) {
+                case "Upcoming due date":
+                    return ['answer_end','compare_end'];
```

Comment:
instead of this can you use
```
                    return function(assignment) {
                        // return rather undefined if empty,
                        // or smaller of answer_end and compare_end if they occur in the future
                        var upcomingDates = [];
                        if (moment(assignment.answer_end).diff(moment()) > 0) {
                            upcomingDates.push(assignment.answer_end);
                        }
                        if (assignment.compare_end && moment(assignment.compare_end).diff(moment()) > 0) {
                            upcomingDates.push(assignment.compare_end);
                        }
                        return _.min(upcomingDates);
                    };
```
This will sort them by whichever is sooner (answer_end or compare_end if defined) while putting assignments already past due date at the bottom

(will need to add `moment` to controller)